plugins/woocommerce/src/Internal/Admin/Agentic/AgenticWebhookPayloadBuilder.php

```diff
@@ -0,0 +1,230 @@
+ *
+ * Builds webhook payloads for the Agentic Commerce Protocol following
+ * the specification for order lifecycle events.
+ *
+ * @since 10.3.0
+ */
+class AgenticWebhookPayloadBuilder {
+	/**
+	 * Build the webhook payload for an order event.
+	 *
+	 * @param string   $event Event type ('order_create' or 'order_update').
+	 * @param WC_Order $order Order object.
+	 * @return array Webhook payload.
+	 */
+	public function build_payload( $event, $order ) {
+		return array(
+			'type' => $event,
+			'data' => $this->build_order_data( $order ),
+		);
+	}
+
+	/**
+	 * Build the order data for the webhook payload.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return array Order data.
+	 */
+	private function build_order_data( $order ) {
+		return array(
+			'type'                => 'order',
+			'checkout_session_id' => $this->get_checkout_session_id( $order ),
+			'permalink_url'       => $this->get_order_permalink( $order ),
+			'status'              => $this->map_order_status( $order->get_status() ),
+			'refunds'             => $this->build_refunds_data( $order ),
+		);
+	}
+
+	/**
+	 * Get the checkout session ID from the order.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return string Checkout session ID.
+	 */
+	private function get_checkout_session_id( $order ) {
+		$session_id = $order->get_meta( '_agentic_checkout_session_id' );
+		return ! empty( $session_id ) ? $session_id : 'checkout_session_' . $order->get_id();
+	}
+
+	/**
+	 * Get the order permalink URL.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return string Order permalink URL.
+	 */
+	private function get_order_permalink( $order ) {
+		// Try to get the order received page URL (customer-facing)
+		$order_received_url = $order->get_checkout_order_received_url();
+
+		if ( ! empty( $order_received_url ) ) {
+			return $order_received_url;
+		}
+
+		// Fallback to admin edit URL if customer URL not available
+		return admin_url( 'post.php?post=' . $order->get_id() . '&action=edit' );
+	}
+
+	/**
+	 * Map WooCommerce order status to ACP status.
+	 *
+	 * ACP statuses: created, manual_review, confirmed, canceled, shipped, fulfilled
+	 *
+	 * @param string $wc_status WooCommerce order status.
+	 * @return string ACP status.
+	 */
+	private function map_order_status( $wc_status ) {
+		// Remove 'wc-' prefix if present.
+		$wc_status = str_replace( 'wc-', '', $wc_status );
+
+		$status_map = array(
+			// WooCommerce status => ACP status.
+			'pending'        => ACPOrderStatus::CREATED,
+			'processing'     => ACPOrderStatus::CONFIRMED,
+			'on-hold'        => ACPOrderStatus::MANUAL_REVIEW,
+			'completed'      => ACPOrderStatus::FULFILLED,
+			'cancelled'      => ACPOrderStatus::CANCELED,
+			'canceled'       => ACPOrderStatus::CANCELED, // Support both spellings.
+			'refunded'       => ACPOrderStatus::FULFILLED, // Refunded orders are still fulfilled.
+			'failed'         => ACPOrderStatus::CANCELED,
+			'checkout-draft' => ACPOrderStatus::CREATED,
+		);
```

Comment:
If we stop removing the `wc-` status above, we can use the stricter constants:
```suggestion
		$status_map = array(
			// WooCommerce status => ACP status.
			OrderStatus::PENDING        => ACPOrderStatus::CREATED,
			OrderStatus::PROCESSING     => ACPOrderStatus::CONFIRMED,
			OrderStatus::ON_HOLD        => ACPOrderStatus::MANUAL_REVIEW,
			OrderStatus::COMPLETED      => ACPOrderStatus::FULFILLED,
			OrderStatus::CANCELLED      => ACPOrderStatus::CANCELED,
			OrderStatus::REFUNDED       => ACPOrderStatus::FULFILLED, // Refunded orders are still fulfilled.
			OrderStatus::FAILED         => ACPOrderStatus::CANCELED,
			OrderStatus::CHECKOUT_DRAFT => ACPOrderStatus::CREATED,
		);
```

> `// Support both spellings.`

Why?

> `// Refunded orders are still fulfilled.`

Yup. Other systems separate the types of statuses, one of the pain points with Woo.